src/main/java/li/cil/oc2/common/bus/device/item/InternetCardItemDevice.java

```diff
@@ -0,0 +1,143 @@
+import li.cil.oc2.api.bus.device.vm.VMDeviceLoadResult;
+import li.cil.oc2.api.bus.device.vm.context.VMContext;
+import li.cil.oc2.api.bus.device.vm.event.VMPausingEvent;
+import li.cil.oc2.api.bus.device.vm.event.VMResumingRunningEvent;
+import li.cil.oc2.common.bus.device.util.IdentityProxy;
+import li.cil.oc2.common.bus.device.util.OptionalAddress;
+import li.cil.oc2.common.bus.device.util.OptionalInterrupt;
+import li.cil.oc2.common.inet.InternetAccess;
+import li.cil.oc2.common.inet.InternetManager;
+import li.cil.oc2.common.serialization.NBTSerialization;
+import li.cil.oc2.common.util.NBTTagIds;
+import li.cil.sedna.device.virtio.VirtIONetworkDevice;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+
+import javax.annotation.Nullable;
+
+@SuppressWarnings("UnstableApiUsage")
+public final class InternetCardItemDevice extends IdentityProxy<ItemStack> implements VMDevice, ItemDevice {
+    private static final String DEVICE_TAG_NAME = "device";
+    private static final String ADDRESS_TAG_NAME = "address";
+    private static final String INTERRUPT_TAG_NAME = "interrupt";
+
+    ///////////////////////////////////////////////////////////////
+    private final InternetAccess internetAccess = new InternetAccessImpl();
+    private final OptionalAddress address = new OptionalAddress();
+    private final OptionalInterrupt interrupt = new OptionalInterrupt();
+    private VirtIONetworkDevice device;
+    private boolean isRunning;
+    private CompoundNBT deviceTag;
+
+    ///////////////////////////////////////////////////////////////
+
+    public InternetCardItemDevice(final ItemStack identity) {
+        super(identity);
+    }
+
+    ///////////////////////////////////////////////////////////////
+
+    @Override
+    public VMDeviceLoadResult mount(final VMContext context) {
+        device = new VirtIONetworkDevice(context.getMemoryMap());
+
+        if (!address.claim(context, device)) {
+            return VMDeviceLoadResult.fail();
+        }
+
+        if (interrupt.claim(context)) {
+            device.getInterrupt().set(interrupt.getAsInt(), context.getInterruptController());
+        } else {
+            return VMDeviceLoadResult.fail();
+        }
+
+        if (deviceTag != null) {
+            NBTSerialization.deserialize(deviceTag, device);
+        }
+
+        context.getEventBus().register(this);
+
+        return VMDeviceLoadResult.success();
+    }
+
+    @Override
+    public void unmount() {
+        suspend();
+        isRunning = false;
+        address.clear();
+        interrupt.clear();
+    }
+
+    @Override
+    public void suspend() {
+        device = null;
+    }
+
+    @Subscribe
+    public void handlePausingEvent(final VMPausingEvent event) {
+        isRunning = false;
+    }
+
+    @Subscribe
+    public void handleResumingRunningEvent(final VMResumingRunningEvent event) {
+        isRunning = true;
+        InternetManager.connect(internetAccess);
```

Comment:
Not sure why I did this. It seems that the point was to stop communicating directly with the `InternetManager` object from the `InternetCardItemDevice` object and use only the `InternetAccess` interface as an intermediary. Otherwise, it turns out that I have to either enter a feedback in the `InternetAccess` interface (`InternetAccess::close` and `InternetAccess::onClose`), or create another method in the `InternetManager` object to change its state.